gitlab/resource_gitlab_project_test.go

```diff
@@ -66,6 +66,50 @@ func TestAccGitlabProject_basic(t *testing.T) {
 					}),
 				),
 			},
+			//Update the project to share the project with a group
+			{
+				Config: testAccGitlabProjectSharedWithGroup(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectExists("gitlab_project.foo", &project),
+					testAccCheckGitlabProjectAttributes(&project, &testAccGitlabProjectExpectedAttributes{
+						Name:                 fmt.Sprintf("foo-%d", rInt),
+						Path:                 fmt.Sprintf("foo.%d", rInt),
+						Description:          "Terraform acceptance tests",
+						IssuesEnabled:        true,
+						MergeRequestsEnabled: true,
+						WikiEnabled:          true,
+						SnippetsEnabled:      true,
+						Visibility:           gitlab.PublicVisibility,
+						SharedWithGroups: []struct {
+							GroupID          int
+							GroupName        string
+							GroupAccessLevel int
+						}{{0, "", 30}},
+					}),
+				),
+			},
+			//Update the project to share the project with more groups
+			{
+				Config: testAccGitlabProjectSharedWithGroup2(rInt),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckGitlabProjectExists("gitlab_project.foo", &project),
+					testAccCheckGitlabProjectAttributes(&project, &testAccGitlabProjectExpectedAttributes{
+						Name:                 fmt.Sprintf("foo-%d", rInt),
+						Path:                 fmt.Sprintf("foo.%d", rInt),
+						Description:          "Terraform acceptance tests",
+						IssuesEnabled:        true,
+						MergeRequestsEnabled: true,
+						WikiEnabled:          true,
+						SnippetsEnabled:      true,
+						Visibility:           gitlab.PublicVisibility,
+						SharedWithGroups: []struct {
+							GroupID          int
+							GroupName        string
+							GroupAccessLevel int
+						}{{0, "", 10}, {0, "", 30}},
+					}),
+				),
+			},
```

Comment:
Maybe add an additional step with testAccGitlabProjectSharedWithGroup to ensure removing a single share also work.